It is not clear what regex syntax Mull accept

### Description

There are at least two Mull options where it accepts regex: `--include-path regex` and `--exclude-path regex
` (see [documentation](https://mull.readthedocs.io/en/latest/CommandLineReference.html)). And it is not clear what syntax should be used.
There are PCRE, POSIX and other syntax. It is worth to specify which one user should use.
I suppose Mull [uses `std::regex`](https://en.cppreference.com/w/cpp/regex).

I have a test project with following files:
```
sergeyb@pony:~/sources/mull_testing$ ls -1
ddd
example
example.c
func1.c
func2.c
main
main.c
Makefile
```
with `grep` I can flter files with prefix `func`, extension `.c` and number after a prefix:
```
sergeyb@pony:~/sources/mull_testing$ ls -1 | grep -E "func[0-9]+\.c"
func1.c
func2.c
```
But Mull ignores such regex:
```
$ mull-cxx --exclude-path="func[0-9]+\.c" ./main
...
[info] Survived mutants (6/6):
/home/sergeyb/sources/mull_testing/func1.c:6:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    a = b + c;
          ^
/home/sergeyb/sources/mull_testing/func2.c:5:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    c = a + b;
          ^
/home/sergeyb/sources/mull_testing/func2.c:7:11: warning: Survived: Replaced - with + [cxx_sub_to_add]
    a = b - c;
          ^
/home/sergeyb/sources/mull_testing/func2.c:8:11: warning: Survived: Replaced - with + [cxx_sub_to_add]
    b = a - b;
          ^
/home/sergeyb/sources/mull_testing/main.c:4:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    a = a + 2;
          ^
/home/sergeyb/sources/mull_testing/main.c:5:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    a = 2 + a;
          ^
...
```

Full output:
```
[debug] Diagnostics: Debug Mode enabled. Debug-level messages will be printed.
[info] Extracting bitcode from executable (threads: 1)

       [################################] 1/1. Finished in 1ms
[info] Loading bitcode files (threads: 3)

       [################################] 3/3. Finished in 11ms
[info] Sanity check run (threads: 1)

       [################################] 1/1. Finished in 4ms
[info] Gathering functions under test (threads: 1)

       [################################] 1/1. Finished in 0ms
[info] Applying function filter: no debug info (threads: 8)

       [################################] 9/9. Finished in 2ms
[info] Applying function filter: file path (threads: 4)

       [################################] 4/4. Finished in 0ms
[info] Instruction selection (threads: 4)

       [################################] 4/4. Finished in 1ms
[info] Searching mutants across functions (threads: 4)

       [################################] 4/4. Finished in 0ms
[info] Applying filter: no debug info (threads: 8)

       [################################] 16/16. Finished in 1ms
[info] Applying filter: file path (threads: 8)

       [################################] 16/16. Finished in 1ms
[info] Applying filter: junk (threads: 8)
[debug] ASTMutationStorage: recording mutation "Add to Sub": /home/sergeyb/sources/mull_testing/main.c:4:11
[debug] ASTMutationStorage: recording mutation "Add to Sub": /home/sergeyb/sources/mull_testing/main.c:5:11

       [##################--------------] 9/16
[debug] ASTMutationStorage: recording mutation "Add to Sub": /home/sergeyb/sources/mull_testing/func2.c:5:11
[debug] ASTMutationStorage: recording mutation "Add to Sub": /home/sergeyb/sources/mull_testing/func1.c:6:11
[debug] ASTMutationStorage: recording mutation "Sub to Add": /home/sergeyb/sources/mull_testing/func2.c:7:11
[debug] ASTMutationStorage: recording mutation "Sub to Add": /home/sergeyb/sources/mull_testing/func2.c:8:11

       [################################] 16/16. Finished in 32ms
[info] Deduplicate mutants (threads: 1)

       [################################] 1/1. Finished in 0ms
[info] Prepare mutations (threads: 1)

       [################################] 1/1. Finished in 0ms
[info] Cloning functions for mutation (threads: 3)

       [################################] 3/3. Finished in 10ms
[info] Removing original functions (threads: 3)

       [################################] 3/3. Finished in 11ms
[info] Redirect mutated functions (threads: 3)

       [################################] 3/3. Finished in 10ms
[info] Applying mutations (threads: 1)

       [################################] 6/6. Finished in 1ms
[info] Compiling original code (threads: 3)
[debug] Emitted object file: /tmp/mull-18a2ca.o for bitcode file: /tmp/mull-e1084d.bc
[debug] Emitted object file: /tmp/mull-53c353.o for bitcode file: /tmp/mull-755eae.bc
[debug] Emitted object file: /tmp/mull-0e3966.o for bitcode file: /tmp/mull-ef1a27.bc

       [################################] 3/3. Finished in 21ms
[info] Link mutated program (threads: 1)
[debug] Link command: clang /tmp/mull-0e3966.o /tmp/mull-18a2ca.o /tmp/mull-53c353.o -o /tmp/mull-d250bb.exe
[info] Compiled executable: /tmp/mull-d250bb.exe

       [################################] 1/1. Finished in 80ms
[info] Warm up run (threads: 1)

       [################################] 1/1. Finished in 2ms
[info] Baseline run (threads: 1)

       [################################] 1/1. Finished in 2ms
[info] Running mutants (threads: 6)

       [################################] 6/6. Finished in 11ms
[info] Survived mutants (6/6):
/home/sergeyb/sources/mull_testing/func1.c:6:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    a = b + c;
          ^
/home/sergeyb/sources/mull_testing/func2.c:5:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    c = a + b;
          ^
/home/sergeyb/sources/mull_testing/func2.c:7:11: warning: Survived: Replaced - with + [cxx_sub_to_add]
    a = b - c;
          ^
/home/sergeyb/sources/mull_testing/func2.c:8:11: warning: Survived: Replaced - with + [cxx_sub_to_add]
    b = a - b;
          ^
/home/sergeyb/sources/mull_testing/main.c:4:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    a = a + 2;
          ^
/home/sergeyb/sources/mull_testing/main.c:5:11: warning: Survived: Replaced + with - [cxx_add_to_sub]
    a = 2 + a;
          ^
[info] Mutation score: 0%
[info] Total execution time: 212ms
```


### Version

Mull: LLVM-based mutation testing
https://github.com/mull-project/mull
Version: 0.10.0-trunk1
Commit: eed5960
Date: 22 Mar 2021
LLVM: 11.0.0

--- a/docs/generated/CLIOptions.rst
+++ b/docs/generated/CLIOptions.rst
@@ -52,20 +52,20 @@
 
 --include-not-covered		Include (but do not run) not covered mutants. Disabled by default
 
---include-path regex		File/directory paths to whitelist (supports regex)
+--include-path regex		File/directory paths to whitelist (supports regex, equivalent to "grep -E")
 
---exclude-path regex		File/directory paths to ignore (supports regex)
+--exclude-path regex		File/directory paths to ignore (supports regex, equivalent to "grep -E")
 
---git-diff-ref git commit		Git branch to run diff against (enables incremental testing)
+--git-diff-ref ref		Git branch, commit, or tag to run diff against (enables incremental testing)
 
---git-project-root git project root		Path to project's Git root (used together with -git-diff-ref)
+--git-project-root path		Path to project's Git root (used together with -git-diff-ref)
 
 --mutators mutator		Choose mutators:
 
     Groups:
       :all:	cxx_all, experimental
 
-      :cxx_all:	cxx_assignment, cxx_increment, cxx_decrement, cxx_arithmetic, cxx_comparison, cxx_boundary, cxx_bitwise
+      :cxx_all:	cxx_assignment, cxx_increment, cxx_decrement, cxx_arithmetic, cxx_comparison, cxx_boundary, cxx_bitwise, cxx_calls
 
       :cxx_arithmetic:	cxx_minus_to_noop, cxx_add_to_sub, cxx_sub_to_add, cxx_mul_to_div, cxx_div_to_mul, cxx_rem_to_div
 
@@ -79,6 +79,8 @@
 
       :cxx_boundary:	cxx_le_to_lt, cxx_lt_to_le, cxx_ge_to_gt, cxx_gt_to_ge
 
+      :cxx_calls:	cxx_remove_void_call, cxx_replace_scalar_call
+
       :cxx_comparison:	cxx_eq_to_ne, cxx_ne_to_eq, cxx_le_to_gt, cxx_lt_to_ge, cxx_ge_to_lt, cxx_gt_to_le
 
       :cxx_const_assignment:	cxx_assign_const, cxx_init_const
@@ -91,7 +93,7 @@
 
       :cxx_logical:	cxx_logical_and_to_or, cxx_logical_or_to_and, cxx_remove_negation
 
-      :experimental:	negate_mutator, remove_void_function_mutator, scalar_value_mutator, replace_call_mutator, cxx_logical
+      :experimental:	negate_mutator, scalar_value_mutator, cxx_logical
 
     Single mutators:
       :cxx_add_assign_to_sub_assign:	Replaces += with -=
@@ -164,6 +166,10 @@
 
       :cxx_remove_negation:	Replaces !a with a
 
+      :cxx_remove_void_call:	Removes calls to a function returning void
+
+      :cxx_replace_scalar_call:	Replaces call to a function with 42
+
       :cxx_rshift_assign_to_lshift_assign:	Replaces >>= with <<=
 
       :cxx_rshift_to_lshift:	Replaces << with >>
@@ -177,10 +183,6 @@
       :cxx_xor_to_or:	Replaces ^ with |
 
       :negate_mutator:	Negates conditionals !x to x and x to !x
-
-      :remove_void_function_mutator:	Removes calls to a function returning void
-
-      :replace_call_mutator:	Replaces call to a function with 42
 
       :scalar_value_mutator:	Replaces zeros with 42, and non-zeros with 0
 

--- a/docs/generated/Mutators.rst
+++ b/docs/generated/Mutators.rst
@@ -68,24 +68,24 @@
 .. |desc33| replace:: Replaces % with /
 .. |op34| replace:: cxx_remove_negation
 .. |desc34| replace:: Replaces !a with a
-.. |op35| replace:: cxx_rshift_assign_to_lshift_assign
-.. |desc35| replace:: Replaces >>= with <<=
-.. |op36| replace:: cxx_rshift_to_lshift
-.. |desc36| replace:: Replaces << with >>
-.. |op37| replace:: cxx_sub_assign_to_add_assign
-.. |desc37| replace:: Replaces -= with +=
-.. |op38| replace:: cxx_sub_to_add
-.. |desc38| replace:: Replaces - with +
-.. |op39| replace:: cxx_xor_assign_to_or_assign
-.. |desc39| replace:: Replaces ^= with \|=
-.. |op40| replace:: cxx_xor_to_or
-.. |desc40| replace:: Replaces ^ with |
-.. |op41| replace:: negate_mutator
-.. |desc41| replace:: Negates conditionals !x to x and x to !x
-.. |op42| replace:: remove_void_function_mutator
-.. |desc42| replace:: Removes calls to a function returning void
-.. |op43| replace:: replace_call_mutator
-.. |desc43| replace:: Replaces call to a function with 42
+.. |op35| replace:: cxx_remove_void_call
+.. |desc35| replace:: Removes calls to a function returning void
+.. |op36| replace:: cxx_replace_scalar_call
+.. |desc36| replace:: Replaces call to a function with 42
+.. |op37| replace:: cxx_rshift_assign_to_lshift_assign
+.. |desc37| replace:: Replaces >>= with <<=
+.. |op38| replace:: cxx_rshift_to_lshift
+.. |desc38| replace:: Replaces << with >>
+.. |op39| replace:: cxx_sub_assign_to_add_assign
+.. |desc39| replace:: Replaces -= with +=
+.. |op40| replace:: cxx_sub_to_add
+.. |desc40| replace:: Replaces - with +
+.. |op41| replace:: cxx_xor_assign_to_or_assign
+.. |desc41| replace:: Replaces ^= with \|=
+.. |op42| replace:: cxx_xor_to_or
+.. |desc42| replace:: Replaces ^ with |
+.. |op43| replace:: negate_mutator
+.. |desc43| replace:: Negates conditionals !x to x and x to !x
 .. |op44| replace:: scalar_value_mutator
 .. |desc44| replace:: Replaces zeros with 42, and non-zeros with 0
 

--- a/lib/Filters/FilePathFilter.cpp
+++ b/lib/Filters/FilePathFilter.cpp
@@ -61,9 +61,9 @@ bool FilePathFilter::shouldSkip(const std::string &sourceFilePath) const {
 std::string FilePathFilter::name() { return "file path"; }
 
 void FilePathFilter::exclude(const std::string &filter) {
-  excludeFilters.push_back(std::regex(filter, std::regex::grep));
+  excludeFilters.emplace_back(filter, std::regex::egrep);
 }
 
 void FilePathFilter::include(const std::string &filter) {
-  includeFilters.push_back(std::regex(filter, std::regex::grep));
+  includeFilters.emplace_back(filter, std::regex::egrep);
 }

--- a/tools/mull-cxx/CLIOptions.cpp
+++ b/tools/mull-cxx/CLIOptions.cpp
@@ -129,14 +129,14 @@ list<MutatorsOptionIndex> tool::Mutators(
 
 list<std::string> tool::ExcludePaths(
   "exclude-path",
-  desc("File/directory paths to ignore (supports regex)"),
+  desc("File/directory paths to ignore (supports regex, equivalent to \"grep -E\")"),
   ZeroOrMore,
   value_desc("regex"),
   cat(MullCXXCategory));
 
 list<std::string> tool::IncludePaths(
   "include-path",
-  desc("File/directory paths to whitelist (supports regex)"),
+  desc("File/directory paths to whitelist (supports regex, equivalent to \"grep -E\")"),
   ZeroOrMore,
   value_desc("regex"),
   cat(MullCXXCategory));
@@ -158,16 +158,16 @@ opt<bool> tool::KeepExecutable(
 
 opt<std::string> tool::GitDiffRef(
     "git-diff-ref",
-    desc("Git branch to run diff against (enables incremental testing)"),
+    desc("Git branch, commit, or tag to run diff against (enables incremental testing)"),
     Optional,
-    value_desc("git commit"),
+    value_desc("ref"),
     cat(MullCXXCategory));
 
 opt<std::string> tool::GitProjectRoot(
     "git-project-root",
     desc("Path to project's Git root (used together with -git-diff-ref)"),
     Optional,
-    value_desc("git project root"),
+    value_desc("path"),
     cat(MullCXXCategory));
 
 opt<bool> tool::IncludeNotCovered(

